### DISCUSSION TITLE
Using subset with gene name in a variable

### DISCUSSION THREAD

**Discussion by cwseidel** [satijalab/seurat#5800] [Q&A] [answered]
How can we use subset if a gene name is within a variable? Previously one could use subset() and supply criteria, such as a gene name, in the form of a variable:

```
mygene <- "Oct4"
subset(obj, subset.names = mygene, low.thresholds = 0)
```

However, the new syntax is to use the gene name directly in an expression:

`subset(obj, Oct4 > 1)`

Putting it in quotes, or in a variable doesn't work:

```
subset(obj, mygene > 1) # doesn't work!
subset(obj, "Oct4" > 1) # doesn't work!
```

One solution is to not use subset(), but rather something of the sort:

`obj[,obj@assays[[1]]@obj[mygene,] > 1]`

I'm aware of one other solution (thanks Malcolm Cook) that seems really obscure:

```
library(rlang)
myGeneVar<-sym(mygene)
inject(subset(x = obj, subset = !!myGeneVar > 1))
```

Both of these seem really obtuse, or esoteric, and I'm just wondering what is the recommended way proceed with subset() if one has a gene name in a variable.

(I'm aware that gene names with dashes with will work with backticks, but that's not relevant here).

**Comment by timoast** [comment, marked as answer]
Please see https://github.com/satijalab/seurat/issues/1053

**Reply by cwseidel** [reply]
Designed for interactive use. Got it. Thanks.
